### PR TITLE
Add new API for adding router in OVN

### LINKED ISSUE #187: Add new API's is Submariner for OVN
As per the enhancement proposal and two new CRDs

type SubmarinerNonGWRouteList struct {
    metav1.TypeMeta `json:",inline"`
    metav1.ListMeta `json:"metadata"`

    Items []SubmarinerNonGWRoute `json:"items"`
}

type SubmarinerRoutePolicySpec struct {
    //Specifies the next hop to reach the remote CIDRs
    NextHop string `json:"nextHop"`

    //Specifies the remote CIDRs available via the next hop
    RemoteCidr []string `json:"remoteCidr"`
}

type SubmarinerGwRouteList struct {
  metav1.TypeMeta `json:",inline"`
  metav1.ListMeta `json:"metadata"`

  Items []SubmarinerGwRoute `json:"items"`
  }

type SubmarinerRoutePolicySpec struct {
    //Specifies the next hop to reach the remote CIDRs
    NextHop string `json:"nextHop"`

    //Specifies the remote CIDRs available via the next hop
    RemoteCidr []string `json:"remoteCidr"`
}

### PR DESCRIPTION
**Pull request by aswinsuryan**
* Added SubmarinerGwRoute API
* Added SubmarinerNonGWRoute API

Fixes: https://github.com/submariner-io/enhancements/issues/187
Depends on :https://github.com/submariner-io/enhancements/pull/163

### CODE REVIEW

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]: What does "dfrt" signify? Should it be "ngwrt"?

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
	RemoteCIDRs []string `json:"remoteCIDRs"`
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]: This doesn't really apply to the spec.

```suggestion
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
```

File: pkg/apis/submariner.io/v1/types.go
Comment (sridhargaddam) [inline pkg/apis/submariner.io/v1/types.go]: Should this be `SubmarinerGwRouteSpec` or are you planning to use the same `SubmarinerRoutePolicySpec` in both SubmarinerGwRoute as well as SubmarinerNonGWRoute?

Comment (aswinsuryan) [reply pkg/apis/submariner.io/v1/types.go]: I thought we will reuse since fields are the same.

**Review by sridhargaddam — APPROVED** [review]

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]: I say we drop he `Submariner` prefix. We don't prepend that with our other CRDs.

```suggestion
type GatewayRoute struct {
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
type GatewayRouteList struct {
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
type NonGatewayRoute struct {
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
type NonGatewayRouteList struct {
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
type RoutePolicySpec struct {
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
	SubmarinerRoutePolicySpec SubmarinerRoutePolicySpec `json:"spec"`
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
	SubmarinerRoutePolicySpec SubmarinerRoutePolicySpec `json:"spec"`
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

**Review by skitt — CHANGES_REQUESTED** [review]
Remember to squash the commits before merging (and feel free to dismiss this review once that’s done).

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]: Let's spell it out for clarity.

```suggestion
type GatewayRoute struct {
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
type GatewayRouteList struct {
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
type NonGatewayRoute struct {
```

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
type NonGatewayRouteList struct {
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/apis/submariner.io/v1/types.go
Comment (tpantelis) [inline pkg/apis/submariner.io/v1/types.go]:
```suggestion
	NextHops []string `json:"nextHops"`
```

**Review by tpantelis — CHANGES_REQUESTED** [review]

File: pkg/apis/submariner.io/v1/types.go
Comment (aswinsuryan) [reply pkg/apis/submariner.io/v1/types.go]: Done.

**Review by tpantelis — APPROVED** [review]

File: pkg/apis/submariner.io/v1/types.go
Comment (sridhargaddam) [inline pkg/apis/submariner.io/v1/types.go:352]: Just thinking out loud, should we use OVN in the names of the struct? 
In the current state, this could give a false impression that this CR will be used in all types of deployments.

Comment (aswinsuryan) [reply pkg/apis/submariner.io/v1/types.go:352]: I think we discussed in the proposal right? Like though this may not be used for any other CNI now , we are not sure if there can be use cases later. Also this is going to be created only for OVN right now so other deployments are not going to see this.

Comment (sridhargaddam) [reply pkg/apis/submariner.io/v1/types.go:352]: Yes we did, but I wanted to throw it open as this is very specific to OVN and not sure if we will use it for other CNIs. Anyways, its okay.

**Comment by aswinsuryan** [comment]
> Remember to squash the commits before merging (and feel free to dismiss this review once that’s done).

Done

**Review by skitt — APPROVED** [review]

**Pull request merged**
